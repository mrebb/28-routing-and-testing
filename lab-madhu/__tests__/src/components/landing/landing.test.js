import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Landing from '../../../../src/components/app/landing/index';

describe('landing component', () => {
  it('is alive at application start', () => {
    let landing = shallow(<Landing/>);
    expect(landing.find('main').exists()).toBeTruthy();
    expect(landing.find('Footer').exists()).toBeTruthy();
  });
});


describe('<Landing/> (Snapshot Test)', () => {
  it('renders right', () => {
    const component = renderer.create(
      <Landing />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

  });
});
