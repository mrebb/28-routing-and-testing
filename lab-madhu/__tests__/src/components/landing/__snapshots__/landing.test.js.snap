// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing/> (Snapshot Test) renders right 1`] = `
Array [
  <main>
    <div>
      <li
        className=""
        role="presentation"
      >
        <a
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="/dashboard"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Dashboard
        </a>
      </li>
    </div>
    <div>
      <h2>
        Welcome to notes app
      </h2>
      <p>
        This app lets you do following actions on notes through dashboard.
        <br />
        <br />
        <li>
          create notes
        </li>
        <li>
          view notes
        </li>
        <li>
          delete notes
        </li>
      </p>
    </div>
  </main>,
  <footer>
    <h3 />
  </footer>,
]
`;
